source/base/DateTime.ooc

```diff
@@ -14,22 +14,233 @@
 * You should have received a copy of the GNU Lesser General Public License
 * along with this software. If not, see <http://www.gnu.org/licenses/>.
 */
-TimeSpan: cover {
-	_ticks: Int64
-	ticks ::= this _ticks
-	init: func@ (=_ticks)
-	kean_base_timeSpan_getTicks: unmangled func -> Int64 { this _ticks }
-	new: unmangled(kean_base_timeSpan_new) static func ~API (ticks: UInt64) -> This { This new(ticks) }
-}
+
+use ooc-base
+import os/Time
 
 DateTime: cover {
 	/* Number of 100 ns intervals since 00.00 1/1/1 */
 	_ticks: UInt64
 	ticks ::= this _ticks
-
 	init: func@ (=_ticks)
+	init: func@ ~fromYearMonthDay (year, month, day: Int) {
+		if (dateIsValid(year, month, day))
+			this _ticks = This dateToTicks(year, month, day)
+		else
+			raise ("invalid input specified for constructor(year,month,day)")
+	}
+	init: func@ ~fromHourMinuteSec (hour, minute, second, millisecond: Int) {
+		if (timeIsValid(hour, minute, second, millisecond))
+			this _ticks = This timeToTicks(hour, minute, second, millisecond)
+		else
+			raise ("invalid input specified for constructor(hour,minute,second)")
+	}
+	init: func@ ~fromDateTime (year, month, day, hour, minute, second, millisecond: Int) {
+		if (dateIsValid(year, month, day) && timeIsValid(hour, minute, second, millisecond))
+			this _ticks = This timeToTicks(hour, minute, second, millisecond) + This dateToTicks(year, month, day)
```

Comment:
Wouldn't this make more sense? `this _ticks = This dateToTicks(year, month, day) + This timeToTicks(hour, minute, second, millisecond)`